jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -3876,8 +3879,15 @@ static void checkBoundsBeginEnd(int begin, int end, int length) {
 	
 	@Override
 	public IntStream chars() {
-		/* Following generic CharSequence method invoking need to be updated with optimized implementation specifically for this class */
-		return CharSequence.super.chars();
+		final int len = lengthInternal();
+		char charBuff[] = new char[len];
+		getCharsNoBoundChecks(0, len, charBuff, 0);
+		int intBuff[] = new int[len];
+		for (int i = 0; i < len; ++i) {
+			intBuff[i] = charBuff[i];
+		}
+		return StreamSupport.intStream(Spliterators.spliterator(intBuff,Spliterator.IMMUTABLE | Spliterator.ORDERED),
+				false);
```

Comment:
Rather than copying the characters, we could return a stream based on either a byte or char spliterator which access the underlying data. As is, an ASCII string requires 6 times additional space and a lot of copying before returning.

Reply:
Keith
Thanks for the idea. I will rework per your suggestion and add some unit tests.
Marking this WIP.